### PR TITLE
E_023_Upgrade_Logging_Logic

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/com/superdevs/HealthOMeter/entity/User.java [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
 
 import javax.persistence.*;
+import java.time.LocalDateTime;
 import java.util.Set;
-/*
-Contact or User - discussion
- */
+
 @Entity(name = "users")
 public class User {
```
Comment (ganzes) [inline src/main/java/com/superdevs/HealthOMeter/entity/User.java:8]: As we discussed on Slack:
LoginData

File: src/main/java/com/superdevs/HealthOMeter/entity/Contact.java [hunk 2]
```diff
@@ -39,6 +36,10 @@
     @Column(name = "updated")
     private LocalDateTime updated;
 
+    @OneToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "user_id")
+    private User user;
+
```
Comment (ganzes) [inline src/main/java/com/superdevs/HealthOMeter/entity/Contact.java:41]: mappedBy = ?
Let's make Contact the main class to handle relations.
And in User class add also @OneToOne regarding Contact, in this version it's only on one side.

File: src/main/java/com/superdevs/HealthOMeter/config/WebSecurityConfig.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
 public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
     @Bean
     public PasswordEncoder getPasswordEncoder() {
-        return new BCryptPasswordEncoder(4);
+        return new BCryptPasswordEncoder();
```
Comment (ganzes) [inline src/main/java/com/superdevs/HealthOMeter/config/WebSecurityConfig.java:17]: Why it was changed? Value (4) is something like password strength?

File: src/main/java/com/superdevs/HealthOMeter/entity/User.java [hunk 4]
```diff
@@ -53,6 +66,22 @@ public void setEnabled(boolean enabled) {
         this.enabled = enabled;
     }
 
+    public LocalDateTime getLastActivity() {
+        return lastActivity;
```
Comment (ganzes) [inline src/main/java/com/superdevs/HealthOMeter/entity/User.java]: Imo shouldn't we be a little more specific? Like: LocalDate.now or something close to that?

File: src/main/java/com/superdevs/HealthOMeter/entity/User.java (hunk 1 above)
Comment (Sephirot92) [reply src/main/java/com/superdevs/HealthOMeter/entity/User.java:8]: Disagree. I take full responsibility for that. 

By default Spring Security is scanning for User and Roles class. If we would like to have LoginData we would have to override SpringClasses. In future we can trhink of it, for now we will move by default.

File: src/main/java/com/superdevs/HealthOMeter/entity/Contact.java (hunk 2 above)
Comment (Sephirot92) [reply src/main/java/com/superdevs/HealthOMeter/entity/Contact.java:41]: Agree. Simple adjustment.

File: src/main/java/com/superdevs/HealthOMeter/config/WebSecurityConfig.java (hunk 3 above)
Comment (Sephirot92) [reply src/main/java/com/superdevs/HealthOMeter/config/WebSecurityConfig.java:17]: Value was for testing purposes. For now we have no users/data/anything. Integer was responsible for BCrypt strenght.

File: src/main/java/com/superdevs/HealthOMeter/entity/User.java (hunk 4 above)
Comment (Sephirot92) [reply src/main/java/com/superdevs/HealthOMeter/entity/User.java]: Hard to tell. @Nebres84 - can I kindly ask for your recommendations?

File: src/main/java/com/superdevs/HealthOMeter/entity/User.java (hunk 1 above)
Comment (ganzes) [reply src/main/java/com/superdevs/HealthOMeter/entity/User.java:8]: Ok, I approve your resolution. This time.

File: src/main/java/com/superdevs/HealthOMeter/entity/Contact.java (hunk 2 above)
Comment (ghost) [reply src/main/java/com/superdevs/HealthOMeter/entity/Contact.java:41]: I created one way binding.

File: src/main/java/com/superdevs/HealthOMeter/config/WebSecurityConfig.java (hunk 3 above)
Comment (ghost) [reply src/main/java/com/superdevs/HealthOMeter/config/WebSecurityConfig.java:17]: Strength - the log rounds to use. The value can range from 4 to 31, but the default is 10.

File: src/main/java/com/superdevs/HealthOMeter/entity/Contact.java (hunk 2 above)
Comment (Sephirot92) [reply src/main/java/com/superdevs/HealthOMeter/entity/Contact.java:41]: @IzabelaRojek 

No one doubts that. Still, even if we are creating OneToOne relation we have to make it "double sided"

https://www.baeldung.com/jpa-one-to-one

As you can see, here we have on both sides Annotations responsible for creating the relation.

If I am wrong, please share some materials for this, I will gladly read them.

Comment (ghost) [reply src/main/java/com/superdevs/HealthOMeter/entity/Contact.java:41]: @Sephirot92 

We can create also one way relationships. If I'm wrong correct me again. Maybe I do not understand it correctly. 

https://docs.oracle.com/cd/E19798-01/821-1841/bnbql/index.html

There is one question. Does the User need to know abot the Contact? 

In our scenario, you are probably right that better approach is creating relation which is double sided. So I'm going to correct it.

**Review by ganzes — APPROVED** [review]

**Pull request merged**